docs/operator_tutorial.md

```diff
@@ -0,0 +1,209 @@
+ userName: 'JohnyAppleseed',
+ line1: '123 Easy St.',
+ line2: '',
+ city: 'Athens',
+ stateProvince: 'GA',
+ postalCode: '30606',
+ country: 'USA'
+ }
+```
+
+Note that all properties now exist at the same level and properties with empty data have been removed.
+
+## Creating Function Arguments with Insert
+
+Now consider this, rather than calling `FS.setUserVars`, you want to call `FS.identify`.  The [identify function](https://developer.fullstory.com/identify) requires not one but two arguments: `identify(uid, object)` where `uid` is a string and the `object` is similar to the one used with `setUserVars`.  To correctly call the function, you need a string value for the first argument and an object for the second.  This use case can be accomplished with the following rule.
+
+```javascript
+{
+  id: "fs-identify-ceddl-user-all",
+  description: "send all CEDDL user properties to FS.identify using the profileID as the FullStory uid",
+  source: "digitalData.user.profile[0]",
+  operators: [ { name: "flatten" } , { name: "insert", select: "profileID" } ],
+  destination: "FS.identify"
+}
+```
+
+The first profile element still needs to be flattened, and this operator remains first.  An **insert** operator is added next.  Each time you add an operator, data is transformed by the previous operator and is passed to the next operator in the sequence.  **Insert** adds a value to create a list of arguments (parameters) that are sent to functions.  The insert operator is configured using a `select` option set to `profileID`.  The `select` option uses selector syntax to find the value at `profileID` in the flattened object and insert it.  The result is a list.
+
+```javascript
+[
+ 'pr-12333211',
+ {
+  profileID: 'pr-12333211',
+  userName: 'JohnyAppleseed',
+  line1: '123 Easy St.',
+  line2: '',
+  city: 'Athens',
+  stateProvince: 'GA',
+  postalCode: '30606',
+  country: 'USA'
+  }
+]
+```
+
+This list is then passed to `FS.identify` to create a function call that looks like `FS.identify('pr-12333211', <theProfileObject>)`.
+
+Congratulations!  By converting data from the data layer using operators, you've essentially called a FullStory API without writing any JavaScript code.
+
+## More About Insert
+
+The insert operator is used quite frequently.  For example the following creates an `FS.event('View Cart', object)` function call using insert's `value` option.  Rather than referencing a property's value from inside an object, the rule creator can specify the *actual* value - "View Cart" cart in this case.  This approach is helpful when you need to name a [Custom Event](https://developer.fullstory.com/custom-events) to provide more clarity to business users.
+
+```javascript
+{
+  id: "fs-event-ceddl-cart",
+  description: "send CEDDL cart's cartID and price properties to FS.event as a 'View Cart' event",
+  source: "digitalData.cart[(cartID,price)]",
+  operators: [ { name: "insert", value: "View Cart" } ],
+  destination: "FS.event"
+}
```

Comment:
would the output of the source + operator combo here be

```
[
'View Cart',
cartId: 'whatever',
price: 10.00
]
```
?

Reply:
yes, correct

Reply:
This is documented in detail in the operator_convert.md doc.